lib/driver.js

```diff
@@ -182,15 +185,38 @@ class YouiEngineDriver extends BaseDriver {
     let iosArgs = {
       javascriptEnabled: true,
     };
-    let iosdriver = new IOSDriver(iosArgs);
+    let driverName = "IOSDriver";
+    if (opts.platformVersion) {
+      let majorVer = opts.platformVersion.toString().split(".")[0];
+      if (parseInt(majorVer, 10) >= 10) {
+        driverName = "XCUITestDriver";
+      }
+    }
+
+    let iosdriver;
+    if (driverName === "XCUITestDriver") {
+      logger.info("Selected driver: XCUITestDriver");
+      iosdriver = new XCUITestDriver(iosArgs);
+    } else {
+      logger.info("Selected driver: IOSDriver");
+      iosdriver = new IOSDriver(iosArgs);
+    }
 
     let caps = {
       platformName: "iOS",
       browserName: "",
-      app: this.caps.app,
-      udid: this.caps.udid,
+      app: opts.app,
+      udid: opts.udid,
```

Comment:
To run on simulator, you should not define udid.
And yes, I did validate I can run on the simulator.
Maybe you can show me your error.